packages/swing-store/src/swingStore.js

```diff
@@ -623,6 +835,53 @@ export function initSwingStore(dirPath = null, options = {}) {
   return makeSwingStore(dirPath, true, options);
 }
 
+/**
+ * @param {SwingStoreExporter} exporter
+ * @param {string | null} [dirPath]
+ * @param {object} options
+ * @returns {Promise<SwingStore>}
+ */
+export async function importSwingStore(exporter, dirPath = null, options = {}) {
+  if (dirPath) {
+    assert.typeof(dirPath, 'string');
+  }
+  const { includeHistorical = false } = options;
+  const store = makeSwingStore(dirPath, true, options);
+  const { kernelStorage } = store;
+  const artifactMetadata = new Map();
+  for (const [key, value] of exporter.getKVData()) {
+    if (key.startsWith('export.')) {
+      artifactMetadata.set(key, value);
+    } else if (value === null) {
+      kernelStorage.kvStore.delete(key);
```

Comment:
Definitely agree with accepting both `null` and `undefined` to mean "this key doesn't exist". Also agree with adding a comment: "most applications will consolidate the exported key/value pairs into a single table, without duplicates or deletions, but in case they merely stash the stream of updates as a list, tolerate duplicate keys (last one wins) and deletions".

Reply:
Actually, the `null` check here was strictly to make TypeScript shut up.  I went ahead and followed the same convention we have in the callback API because that seemed like a safe and benign thing to do, but in reality this case should never happen.

Reply:
It actually can happen, if the streaming exporter is used, but the host app is being too dumb and sets to `null` instead of deleting the exported entry. Since it doesn't cost much, let's keep handling this case, but with the more flexible `== null`.

Reply:
> won't `== null` result in a bitchy complaint from eslint?

I convinced MarkM that `== null` was entirely acceptable given the language has this null-ish concept now, and the eslint rules have been relaxed.

Reply:
That's still subtle enough that I'm going to document the use with a comment.